core/src/test/java/org/apache/iceberg/TestCreateTransaction.java

```diff
@@ -267,10 +267,6 @@ public void testCreateDetectsUncommittedChangeOnCommit() throws IOException {
         TestTables.metadataVersion("uncommitted_change"));
 
     txn.updateProperties().set("test-property", "test-value"); // not committed
-
-    AssertHelpers.assertThrows("Should reject commit when last operation has not committed",
```

Comment:
You may want to remove the entire unit test since this assertion is mainly logic for this unit test.

This introduces a behaviour change which could ignore the operations that haven't been commit yet. Not sure whether the downstream client will be impacted by this or not.